spec/octopus/serverspec/type/octopus_deploy_environment_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+require 'webmock/rspec'
+
+describe OctopusDeployEnvironment do
+
+    let(:runner) { double ("runner")}
+
+    it "throws if `serverUrl` not supplied" do
+        expect { OctopusDeployEnvironment.new(nil, "someapikey", "my new environment") }.
+            to raise_error(/serverUrl/)
+    end
+
+    it "throws if `apiKey` not supplied" do
+        expect { OctopusDeployEnvironment.new("https://someserver.com", nil, "my new environment") }.
+            to raise_error(/apiKey/)
+    end
+
+    it "throws if `environmentname` not supplied" do
+        expect { OctopusDeployEnvironment.new("https://someserver.com", "API-kllkjhasdkljhasdfkjsafd", nil) }.
+            to raise_error(/environment_name/)
+    end
+
+    examplejsonpath = 'spec/octopus/serverspec/json/envfound.json'
+    examplejson = File.open(examplejsonpath)
```

Comment:
Consider something like `json_with_environment` and similar for the other variables.
Also not sure if you need the path variables - you could inline them.